src/Components/Users/SkillsSlideOver.tsx

```diff
@@ -62,36 +56,32 @@ export default ({ show, setShow, username }: IProps) => {
       }
       setSelectedSkill(null);
       setIsLoading(false);
-      fetchSkills(username);
+      await refetchUserSkills();
     },
-    [dispatch, fetchSkills]
+    [refetchUserSkills]
   );
 
   const removeSkill = useCallback(
     async (username: string, skillId: string) => {
-      const res = await dispatch(deleteUserSkill(username, skillId));
+      const { res } = await request(routes.deleteUserSkill, {
+        pathParams: { username, id: skillId },
+      });
       if (res?.status !== 204) {
         Notification.Error({
           msg: "Error while unlinking skill",
         });
       }
       setDeleteSkill(null);
-      fetchSkills(username);
+      await refetchUserSkills();
     },
-    [dispatch, fetchSkills]
+    [refetchUserSkills]
   );
 
-  useEffect(() => {
-    setIsLoading(true);
-    if (username) fetchSkills(username);
-    setIsLoading(false);
-  }, [username, fetchSkills]);
-
   const authorizeForAddSkill = useIsAuthorized(
     AuthorizeFor(["DistrictAdmin", "StateAdmin"])
   );
 
-  const hasSkills = useMemo(() => skills.length > 0, [skills]);
+  const hasSkills = useMemo(() => skills?.results?.length || 0 > 0, [skills]);
```

Comment:
`useMemo` should be used for more expensive computes only. The overhead of the `useMemo` would make the performance bad for very simple computes like these.

```suggestion
  const hasSkills = skills?.results?.length || 0 > 0;
```